common/fe-ste-models.go

```diff
@@ -145,6 +145,43 @@ func (d DeleteSnapshotsOption) ToDeleteSnapshotsOptionType() azblob.DeleteSnapsh
 	return azblob.DeleteSnapshotsOptionType(strings.ToLower(d.String()))
 }
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+var EPermanentDeleteOption = PermanentDeleteOption(3) // Default to "None"
+
+type PermanentDeleteOption uint8
+
+func (PermanentDeleteOption) Snapshots() PermanentDeleteOption { return PermanentDeleteOption(0) }
```

Comment:
I think making these bitflags would be better here because...

Reply:
I see where Adele is coming from. I think it'll make comparisons easier instead of writing switch statement every time. 
None = 0,
Snapshot = 1
Version = 2
SnapshotAndVersion = Snapshot|Version = 0b01 | 0b10 = ob11 = 3

We'll not have to create 3 variables in the code that way.

Reply:
```golang
(o PermanentDeleteOption) DeleteSnapshot() {
    return o & EPermanentDeleteOption.Snapshot();
}

(o PermanentDeleteOption) DeleteVersion() {
    return o & EPermanentDeleteOption.Version();
}
```

These functions can act as flags anywhere you may want.